packages/vite/src/node/plugins/resolve.ts

```diff
@@ -126,6 +126,7 @@ export function resolvePlugin(resolveOptions: InternalResolveOptions): Plugin {
   } = resolveOptions
 
   const { target: ssrTarget, noExternal: ssrNoExternal } = ssrConfig ?? {}
+  const resolvedCache = new Map()
```

Comment:
Is it possible to type the Map?

Reply:
I am not sure whether `ResolvedConfig`'s shallow value will change or not. My first reaction is using `JSON.stringify(ResolvedConfig)` as a cache key.

Reply:
We are already using the same pattern for all other caches (as the one linked in my last message). `ResolvedConfig` can be considered immutable once the dev server is running or the rollup build started (the only case where it could be modified is in the `configResolved` hooks in some plugins, but is more like a hack). If the `ResolvedConfig` changes, all values should be invalidated so having the Cache against it works well. It is also done like this because the plugins are cloned when the server restarts or there is a rebuild, so we need this general invalidation.